### ISSUE TITLE
Right-clicking tray icon early during startup causes crash

### ISSUE THREAD

**Issue by canton7** [canton7/SyncTrayzor#257]
I think, at least.

```
2016-05-03 21:54:11.5531 #56524 [Info] SyncTrayzor.Services.Conflicts.ConflictFileWatcher: Refreshing conflicted files. Found 0 from 0 options 
2016-05-03 21:54:15.6758 #56524 [Error] SyncTrayzor.Bootstrapper: An unhandled AppDomain exception occurred. Terminating: True Stylet.Xaml.ActionNotSetException View.ActionTarget not set on control Hardcodet.Wpf.TaskbarNotification.TaskbarIcon (method DoubleClick). This probably means the control hasn't inherited it from a parent, e.g. because a ContextMenu or Popup sits in the visual tree. You will need so set 's:View.ActionTarget' explicitly. See the wiki section "Actions" for more details. Void AssertTargetSet()    at Stylet.Xaml.ActionBase.AssertTargetSet() in c:\projects\stylet\Stylet\Xaml\ActionBase.cs:line 172
   at Stylet.Xaml.CommandAction.Execute(Object parameter) in c:\projects\stylet\Stylet\Xaml\CommandAction.cs:line 155
   at Hardcodet.Wpf.TaskbarNotification.Util.ExecuteIfEnabled(ICommand command, Object commandParameter, IInputElement target) in e:\Repositories\Open Source\evolve.net\WPF\NotifyIcon\Source\NotifyIconWpf\Util.cs:line 273
   at Hardcodet.Wpf.TaskbarNotification.TaskbarIcon.RaiseTrayMouseDoubleClickEvent() in e:\Repositories\Open Source\evolve.net\WPF\NotifyIcon\Source\NotifyIconWpf\TaskbarIcon.Declarations.cs:line 1110
   at Hardcodet.Wpf.TaskbarNotification.TaskbarIcon.OnMouseEvent(MouseEvent me) in e:\Repositories\Open Source\evolve.net\WPF\NotifyIcon\Source\NotifyIconWpf\TaskbarIcon.cs:line 373
   at Hardcodet.Wpf.TaskbarNotification.Interop.WindowMessageSink.ProcessWindowMessage(UInt32 msg, IntPtr wParam, IntPtr lParam) in e:\Repositories\Open Source\evolve.net\WPF\NotifyIcon\Source\NotifyIconWpf\Interop\WindowMessageSink.cs:line 254
   at Hardcodet.Wpf.TaskbarNotification.Interop.WindowMessageSink.OnWindowMessageReceived(IntPtr hwnd, UInt32 messageId, IntPtr wparam, IntPtr lparam) in e:\Repositories\Open Source\evolve.net\WPF\NotifyIcon\Source\NotifyIconWpf\Interop\WindowMessageSink.cs:line 217 Stylet.Xaml.ActionNotSetException: View.ActionTarget not set on control Hardcodet.Wpf.TaskbarNotification.TaskbarIcon (method DoubleClick). This probably means the control hasn't inherited it from a parent, e.g. because a ContextMenu or Popup sits in the visual tree. You will need so set 's:View.ActionTarget' explicitly. See the wiki section "Actions" for more details.
   at Stylet.Xaml.ActionBase.AssertTargetSet() in c:\projects\stylet\Stylet\Xaml\ActionBase.cs:line 172
   at Stylet.Xaml.CommandAction.Execute(Object parameter) in c:\projects\stylet\Stylet\Xaml\CommandAction.cs:line 155
   at Hardcodet.Wpf.TaskbarNotification.Util.ExecuteIfEnabled(ICommand command, Object commandParameter, IInputElement target) in e:\Repositories\Open Source\evolve.net\WPF\NotifyIcon\Source\NotifyIconWpf\Util.cs:line 273
   at Hardcodet.Wpf.TaskbarNotification.TaskbarIcon.RaiseTrayMouseDoubleClickEvent() in e:\Repositories\Open Source\evolve.net\WPF\NotifyIcon\Source\NotifyIconWpf\TaskbarIcon.Declarations.cs:line 1110
   at Hardcodet.Wpf.TaskbarNotification.TaskbarIcon.OnMouseEvent(MouseEvent me) in e:\Repositories\Open Source\evolve.net\WPF\NotifyIcon\Source\NotifyIconWpf\TaskbarIcon.cs:line 373
   at Hardcodet.Wpf.TaskbarNotification.Interop.WindowMessageSink.ProcessWindowMessage(UInt32 msg, IntPtr wParam, IntPtr lParam) in e:\Repositories\Open Source\evolve.net\WPF\NotifyIcon\Source\NotifyIconWpf\Interop\WindowMessageSink.cs:line 254
   at Hardcodet.Wpf.TaskbarNotification.Interop.WindowMessageSink.OnWindowMessageReceived(IntPtr hwnd, UInt32 messageId, IntPtr wparam, IntPtr lparam) in e:\Repositories\Open Source\evolve.net\WPF\NotifyIcon\Source\NotifyIconWpf\Interop\WindowMessageSink.cs:line 217
```

**Comment by fl0wfr** [comment]
This issue is still there, I just installed the app on a Windows 10 tablet and I got the same issue...

**Comment by EchterAgo** [comment]
Yup, still there, also just had the same issue.

**Comment by canton7** [comment]
Can you post the crash message? I want to check that it is actually the same issue.

**Comment by EchterAgo** [comment]
```
2017-10-24 13:42:28.8726 #53428 [Error] SyncTrayzor.Bootstrapper: An unhandled AppDomain exception occurred. Terminating: True Stylet.Xaml.ActionNotSetException View.ActionTarget not set on control Hardcodet.Wpf.TaskbarNotification.TaskbarIcon (method DoubleClick). This probably means the control hasn't inherited it from a parent, e.g. because a ContextMenu or Popup sits in the visual tree. You will need so set 's:View.ActionTarget' explicitly. See the wiki section "Actions" for more details. Void AssertTargetSet()    at Stylet.Xaml.ActionBase.AssertTargetSet() in c:\projects\stylet\Stylet\Xaml\ActionBase.cs:line 172
   at Stylet.Xaml.CommandAction.Execute(Object parameter) in c:\projects\stylet\Stylet\Xaml\CommandAction.cs:line 155
   at Hardcodet.Wpf.TaskbarNotification.Util.ExecuteIfEnabled(ICommand command, Object commandParameter, IInputElement target) in c:\git\OSS\notifyicon-wpf\Hardcodet.NotifyIcon.Wpf\Source\NotifyIconWpf\Util.cs:line 273
   at Hardcodet.Wpf.TaskbarNotification.TaskbarIcon.RaiseTrayMouseDoubleClickEvent() in c:\git\OSS\notifyicon-wpf\Hardcodet.NotifyIcon.Wpf\Source\NotifyIconWpf\TaskbarIcon.Declarations.cs:line 1137
   at Hardcodet.Wpf.TaskbarNotification.TaskbarIcon.OnMouseEvent(MouseEvent me) in c:\git\OSS\notifyicon-wpf\Hardcodet.NotifyIcon.Wpf\Source\NotifyIconWpf\TaskbarIcon.cs:line 389
   at Hardcodet.Wpf.TaskbarNotification.Interop.WindowMessageSink.ProcessWindowMessage(UInt32 msg, IntPtr wParam, IntPtr lParam) in c:\git\OSS\notifyicon-wpf\Hardcodet.NotifyIcon.Wpf\Source\NotifyIconWpf\Interop\WindowMessageSink.cs:line 252
   at Hardcodet.Wpf.TaskbarNotification.Interop.WindowMessageSink.OnWindowMessageReceived(IntPtr hwnd, UInt32 messageId, IntPtr wparam, IntPtr lparam) in c:\git\OSS\notifyicon-wpf\Hardcodet.NotifyIcon.Wpf\Source\NotifyIconWpf\Interop\WindowMessageSink.cs:line 216 Stylet.Xaml.ActionNotSetException: View.ActionTarget not set on control Hardcodet.Wpf.TaskbarNotification.TaskbarIcon (method DoubleClick). This probably means the control hasn't inherited it from a parent, e.g. because a ContextMenu or Popup sits in the visual tree. You will need so set 's:View.ActionTarget' explicitly. See the wiki section "Actions" for more details.
   at Stylet.Xaml.ActionBase.AssertTargetSet() in c:\projects\stylet\Stylet\Xaml\ActionBase.cs:line 172
   at Stylet.Xaml.CommandAction.Execute(Object parameter) in c:\projects\stylet\Stylet\Xaml\CommandAction.cs:line 155
   at Hardcodet.Wpf.TaskbarNotification.Util.ExecuteIfEnabled(ICommand command, Object commandParameter, IInputElement target) in c:\git\OSS\notifyicon-wpf\Hardcodet.NotifyIcon.Wpf\Source\NotifyIconWpf\Util.cs:line 273
   at Hardcodet.Wpf.TaskbarNotification.TaskbarIcon.RaiseTrayMouseDoubleClickEvent() in c:\git\OSS\notifyicon-wpf\Hardcodet.NotifyIcon.Wpf\Source\NotifyIconWpf\TaskbarIcon.Declarations.cs:line 1137
   at Hardcodet.Wpf.TaskbarNotification.TaskbarIcon.OnMouseEvent(MouseEvent me) in c:\git\OSS\notifyicon-wpf\Hardcodet.NotifyIcon.Wpf\Source\NotifyIconWpf\TaskbarIcon.cs:line 389
   at Hardcodet.Wpf.TaskbarNotification.Interop.WindowMessageSink.ProcessWindowMessage(UInt32 msg, IntPtr wParam, IntPtr lParam) in c:\git\OSS\notifyicon-wpf\Hardcodet.NotifyIcon.Wpf\Source\NotifyIconWpf\Interop\WindowMessageSink.cs:line 252
   at Hardcodet.Wpf.TaskbarNotification.Interop.WindowMessageSink.OnWindowMessageReceived(IntPtr hwnd, UInt32 messageId, IntPtr wparam, IntPtr lparam) in c:\git\OSS\notifyicon-wpf\Hardcodet.NotifyIcon.Wpf\Source\NotifyIconWpf\Interop\WindowMessageSink.cs:line 216
```

**Comment by EchterAgo** [comment]
Pretty much the same, except for paths.

**Comment by EchterAgo** [comment]
Using latest SyncTrayzor downloaded today.

**Comment by canton7** [comment]
This should now be fixed: with v1.1.26, the tray icon is hidden until start has reached the point where it can handle interactions
